ssdp.md

```diff
@@ -0,0 +1,135 @@
+    USN: advertisement UUID
+    NT: urn:example-org:service:osp:1
+    FRIENDLY-NAME.example.org: My Presentation Display
+    PRESENTATION-ENDPOINT.example.org: 192.168.1.100:3000
+    ```
+
+* A controller sends the following SSDP search message to the multicast address. The new header `PRESENTATION-URLS.example.org` allows the controller to send the presentation URLs to the display. 
+
+    ```
+    M-SEARCH * HTTP/1.1
+    HOST: 239.255.255.250:1900
+    MAN: "ssdp:discover"
+    MX: seconds to delay response
+    ST: urn:example-org:service:osp:1
+    PRESENTATION-URLS.example.org: http://example.com/foo.html, http://example.com/bar.html
+    ```
+* A display that can open one of the URLs replies (unicast) with the following SSDP message. The new SSDP header `SUPPORTED-URLS.example.org` contains the URLs the receiver can open from the list of the URLs `PRESENTATION-URLS.example.org` sent in the search SSDP message.
+
+    ```
+    HTTP/1.1 200 OK
+    CACHE-CONTROL: max-age = seconds until advertisement expires
+    DATE: when response was generated
+    LOCATION: URL of the device description
+    SERVER: OS/version UPnP/1.0 product/version
+    USN: advertisement UUID
+    ST: urn:example-org:service:osp:1
+    FRIENDLY-NAME.example.org: My Presentation Display
+    PRESENTATION-ENDPOINT.example.org: 192.168.1.100:3000
+    SUPPORTED-URLS.example.org: http://example.com/foo.html
+    ```
+* The display sends the following SSDP message when the receiver service in not more available. There are no new headers used in the byebye message.
+
+    ```
+    NOTIFY * HTTP/1.1
+    HOST: 239.255.255.250:1900
+    NT: urn:example-org:service:osp:1
+    NTS: ssdp:byebye
+    USN: uuid:advertisement UUID
+    ```
+
+### Remote Playback API: Monitor Display Availability
+TODO (it should be very similar to the Presentation API display availability but for capabilities other parameters related media formats, codecs, encryptions need to be considered)
+
+## Non-functional Requirements
+
+### Reliability
+
+UPnP recommend due to the unreliable nature of UDP to send the SSDP messages more than once (and max three times to avoid network congestion) with some delay of few hundred milliseconds. In addition, the device must re-send its advertisements periodically prior to expiration of the duration specified in
+the `CACHE-CONTROL` header (minimum value is 1800s).  
+
+### Latency of device discovery / device removal
+
+New devices added or removed can be immediately detected if the controller listens to the multicast address for "ssdp:alive" and "ssdp:byebye" messages. For search requests the latency depends on the `MX` SSDP header which contains the maximum wait time in seconds. According to UPnP specification, it must be greater than or equal to 1 and should be less than 5 inclusive. SSDP responses should be delayed a random duration between 0 and the value of `MX` to balance load for the controller when it processes responses.
```

Comment:
It depends from the implementation of SSDP on the controller. For the case "TV shut off and sends byebye" and the controller is listening on the multicast address "239.255.255.250:1900" the device removal will be detected as soon as the byebye message is received and parsed (It will take maybe few milliseconds).  For the case "shut off without any notification" and the controller is listening on the "239.255.255.250:1900" the controller can only detect the device removal after the last advertisement received from that device expires (`CACHE-CONTROL` header contains the expiration in seconds).  

 If the controller is not listening on the multicast address "239.255.255.250:1900" there is no difference between the two cases you mentioned and the device removal can only be detected after the controller sends the next SEARCH message and compare the response messages with the old list of available devices.